service/api.go

```diff
@@ -667,14 +667,20 @@ func (as *APIServer) validateProjectConfig(w http.ResponseWriter, r *http.Reques
 		errs = append(errs, validationErr)
 	}
 
-	errs = append(errs, validator.CheckYamlStrict(input.ProjectYaml)...)
-	errs = append(errs, validator.CheckProjectSyntax(project, input.IncludeLong)...)
-	errs = append(errs, validator.CheckProjectSemantics(project)...)
+	errs = append(errs, validator.CheckProjectErrors(project, input.IncludeLong)...)
 
-	if len(errs) > 0 {
-		if input.Quiet {
-			errs = errs.AtLevel(validator.Error)
+	if input.Quiet {
+		errs = errs.AtLevel(validator.Error)
+		if len(errs) > 0 {
+			gimlet.WriteJSONError(w, errs)
+		} else {
+			gimlet.WriteJSON(w, validator.ValidationErrors{})
 		}
+		return
+	}
+
+	errs = append(errs, validator.CheckProjectWarnings(project, input.ProjectYaml)...)
```

Comment:
We can simplify this section with 
```
if input.Quiet {
    errs = append(errs, validator.CheckProjectWarnings(project, input.ProjectYaml)...)
} else {
    errs = errs.AtLevel(validator.Error)
}
``` 
and then use the same `if len(errs) > 0` logic, rather than having it twice